src/daft-io/src/lib.rs

```diff
@@ -367,6 +384,9 @@ pub fn parse_url(input: &str) -> Result<(SourceType, Cow<'_, str>)> {
                 msg: "Could not convert expanded path to string".to_string(),
             });
     }
+    if input.starts_with("hf://") {
+        return Ok((SourceType::HF, Cow::Borrowed(input)));
+    }
```

Comment:
Is there a reason for having this as well as the match on `scheme` on like 409?

Reply:
just tested without this, and it  seems to work fine, will remove!